clouddriver-web/src/main/groovy/com/netflix/spinnaker/clouddriver/controllers/SearchController.groovy

```diff
@@ -45,15 +48,28 @@ class SearchController {
    */
   @PreAuthorize("@fiatPermissionEvaluator.storeWholePermission()")
   @RequestMapping(value = '/search', method = RequestMethod.GET)
-  List<SearchResultSet> search(SearchQueryCommand q) {
+  List<SearchResultSet> search(@RequestParam(value = "q", defaultValue = "", required = false) String query,
+                               @RequestParam(value = "type") List<String> type,
+                               @RequestParam(value = "platform", required = false) String platform,
+                               @RequestParam(value = "pageSize", defaultValue = "10000", required = false) int pageSize,
+                               @RequestParam(value = "page", defaultValue = "1", required = false) int page,
+                               HttpServletRequest httpServletRequest) {
+
+    def filters = httpServletRequest.getParameterNames().findAll { String parameterName ->
+      !["q", "type", "platform", "pageSize", "page"].contains(parameterName)
+    }.collectEntries { String parameterName ->
+      [parameterName, httpServletRequest.getParameter(parameterName)]
+    }
 
-    log.info("Fetching search results for ${q.q}, platform: ${q.platform}, type: ${q.type}, pageSize: ${q.pageSize}, pageNumber: ${q.page}, filter: ${q.filter}")
+    def searchQuery =
```

Comment:
Obviously not necessary, but I've been trying to avoid duck typing whenever I know what the thing is. Makes a future conversion to Java a lot easier.